python/cudf/cudf/core/frame.py

```diff
@@ -1409,12 +1409,23 @@ def searchsorted(
         if len(values) != len(self._data):
             raise ValueError("Mismatch number of columns to search for.")
 
+        common_dtype_list = [
+            find_common_type([col.dtype, val.dtype])
+            for col, val in zip(self._columns, values)
+        ]
         sources = [
             col
-            if is_dtype_equal(col.dtype, val.dtype)
-            else col.astype(val.dtype)
-            for col, val in zip(self._columns, values)
+            if is_dtype_equal(col.dtype, common_dtype)
+            else col.astype(common_dtype)
+            for col, common_dtype in zip(self._columns, common_dtype_list)
         ]
+        values = [
+            val
+            if is_dtype_equal(val.dtype, common_dtype)
+            else val.astype(common_dtype)
+            for val, common_dtype in zip(values, common_dtype_list)
```

Comment:
This allows:
```
s = cudf.Series([1, 2, 3], index=[1, 2, 3])
s.loc["1":"3"]  # same as if we wrote s.loc[1:3]
```
to succeed, even though `s.loc["1"]` and `s.loc["3"]` raise `KeyError`. But I note that pandas does the same thing here, so perhaps this is safe (@mroeschke is this a bug?). If I make a series:
```
import pandas as pd
s = pd.Series([1, 2, "1"])
s.sort_values() # raise ValueError
# But
s = pd.Series([1, 2])
s.searchsorted("1") # => array([0]), whereas I was expecting ValueError
```

Reply:
Interesting. I think we are mirroring this behavior from numpy but I think pandas should be raising a ValueError here too

```python
In [4]: import numpy as np

In [5]: arr = np.array([1, 2])

In [6]: arr.searchsorted("1")
Out[6]: 0
```

Reply:
@mroeschke Would it be possible for you to open a pandas issue to track this? I can link it in our code for future change/reference.

Reply:
Sure, https://github.com/pandas-dev/pandas/issues/54668

Reply:
Thanks @mroeschke ! 

@wence- I added a TODO to update our code behavior in future based on how https://github.com/pandas-dev/pandas/issues/54668 progresses. Let me know if you think that's okay.